src/mono/netcore/System.Private.CoreLib/src/System/Threading/ThreadPool.Browser.Mono.cs

```diff
@@ -10,26 +10,49 @@
 
 namespace System.Threading
 {
-    public sealed class RegisteredWaitHandle : MarshalByRefObject
+    public sealed partial class Thread
```

Comment:
I think this should better be in Tread.Mono.cs

Reply:
Yea it's not the ideal place for it. The build that uses the portable thread pool gets the same implementation from ThreadPool.Portable.cs. I need a coreclr-specific version for now and an "everything else" version, eventually coreclr would also use the common implementation. So perhaps a better place would be in the shared Thread.cs under a `#if !CORECLR` (which can eventually be removed).